applylib/applyset/results.go

```diff
@@ -22,70 +22,86 @@ import (
 	"k8s.io/klog/v2"
 )
 
+type ObjectStatus struct {
+	GVK           schema.GroupVersionKind
+	NameNamespace types.NamespacedName
+	IsHealthy     bool
+	Status        string
+	Message       string
+}
+
 // ApplyResults contains the results of an Apply operation.
 type ApplyResults struct {
-	total             int
-	applySuccessCount int
-	applyFailCount    int
-	pruneSuccessCount int
-	pruneFailCount    int
-	healthyCount      int
-	unhealthyCount    int
+	Total             int
+	ApplySuccessCount int
+	ApplyFailCount    int
+	PruneSuccessCount int
+	PruneFailCount    int
+	HealthyCount      int
+	UnhealthyCount    int
+	Objects           []ObjectStatus
```

Comment:
So these were previously unexported so that we could change the implementation and instead offer higher-level methods (AllApplied, All healthy).  What signals do we _want_ to expose?  And which objects appear in Objects?  One answer could be that we every object in the manifest appears in Objects, and then we populate Applied, Is healthy etc at the object level.  But what about objects we are pruning - should they appear in Objects?

I guess the question is ... what do you want to do with this information?  Do we have to commit to exposing everything, or can we make more incremental changes?

Reply:
We mostly want just the objects. Updated the PR to reflect that. Not exposing the counts anymore. Just the list of object status.